scripts/main.js

```diff
@@ -1,32 +1,37 @@
 $("#new-app-btn").on("click", function(){
-$("#apps-container").hide();
-$("#create-app-container").show();
-$("#detail-app-container").hide();
-	//create an new app, when a user click 'create button'
-var createBtn = document.querySelector("#choose-create");
-if(createBtn){
-  createBtn.addEventListener("click",function(event){
-			//create app array for exisiting apps
-	var appname = document.getElementsByTagName("input")[0].value;
-	var app = { name: appname};
-	if(app.name){
-		applist.add ({
-		  name : appname
-		})
-	makeAppList();
-	showAppDetail(app.id);
```

Comment:
the problem here was that app.id was undefined, because `applist.add()` returns a promise. So you have to add a `.then()` callback which gets an `app` argument which then has the `app.id` defined